metrics-healthchecks/src/main/java/com/codahale/metrics/health/HealthCheck.java

```diff
@@ -1,15 +1,20 @@
 package com.codahale.metrics.health;
 
+import java.util.Collections;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
 /**
  * A health check for a component of your application.
  */
 public abstract class HealthCheck {
     /**
-     * The result of a {@link HealthCheck} being run. It can be healthy (with an optional message)
-     * or unhealthy (with either an error message or a thrown exception).
+     * The result of a {@link HealthCheck} being run. It can be healthy (with an optional message and optional details)
+     * or unhealthy (with either an error message or a thrown exception and optional details).
      */
     public static class Result {
-        private static final Result HEALTHY = new Result(true, null, null);
+        private static final Result HEALTHY = new ResultBuilder(true).build();
```

Comment:
The idea was to use the same Builder everywhere but and have only one single constructor which takes the builder as parameter. I think it give us more scalability and a better encapsulation. For example if we would like to add a new specific field, we can do it and add a default value in the builder. So if you don't use it, it will have the default value without the need to change this code.